Sources/Examples/Echo/Model/echo.grpc.swift

```diff
@@ -172,6 +172,173 @@ public final class Echo_EchoClient: Echo_EchoClientProtocol {
   }
 }
 
+#if compiler(>=5.5)
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public protocol Echo_EchoAsyncClientProtocol: GRPCClient {
+  var serviceName: String { get }
+  var interceptors: Echo_EchoClientInterceptorFactoryProtocol? { get }
+
+  func makeGetCall(
+    _ request: Echo_EchoRequest,
+    callOptions: CallOptions?
+  ) -> GRPCAsyncUnaryCall<Echo_EchoRequest, Echo_EchoResponse>
+
+  func makeExpandCall(
+    _ request: Echo_EchoRequest,
+    callOptions: CallOptions?
+  ) -> GRPCAsyncServerStreamingCall<Echo_EchoRequest, Echo_EchoResponse>
+
+  func makeCollectCall(
+    callOptions: CallOptions?
+  ) -> GRPCAsyncClientStreamingCall<Echo_EchoRequest, Echo_EchoResponse>
+
+  func makeUpdateCall(
+    callOptions: CallOptions?
+  ) -> GRPCAsyncBidirectionalStreamingCall<Echo_EchoRequest, Echo_EchoResponse>
+}
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+extension Echo_EchoAsyncClientProtocol {
+  public var serviceName: String {
+    return "echo.Echo"
+  }
+
+  public var interceptors: Echo_EchoClientInterceptorFactoryProtocol? {
+    return nil
+  }
+
+  public func makeGetCall(
+    _ request: Echo_EchoRequest,
+    callOptions: CallOptions? = nil
+  ) -> GRPCAsyncUnaryCall<Echo_EchoRequest, Echo_EchoResponse> {
+    return self.makeAsyncUnaryCall(
+      path: "/echo.Echo/Get",
+      request: request,
+      callOptions: callOptions ?? self.defaultCallOptions
+    )
+  }
+
+  public func makeExpandCall(
+    _ request: Echo_EchoRequest,
+    callOptions: CallOptions? = nil
+  ) -> GRPCAsyncServerStreamingCall<Echo_EchoRequest, Echo_EchoResponse> {
+    return self.makeAsyncServerStreamingCall(
+      path: "/echo.Echo/Expand",
+      request: request,
+      callOptions: callOptions ?? self.defaultCallOptions
+    )
+  }
+
+  public func makeCollectCall(
+    callOptions: CallOptions? = nil
+  ) -> GRPCAsyncClientStreamingCall<Echo_EchoRequest, Echo_EchoResponse> {
+    return self.makeAsyncClientStreamingCall(
+      path: "/echo.Echo/Collect",
+      callOptions: callOptions ?? self.defaultCallOptions
+    )
+  }
+
+  public func makeUpdateCall(
+    callOptions: CallOptions? = nil
+  ) -> GRPCAsyncBidirectionalStreamingCall<Echo_EchoRequest, Echo_EchoResponse> {
+    return self.makeAsyncBidirectionalStreamingCall(
+      path: "/echo.Echo/Update",
+      callOptions: callOptions ?? self.defaultCallOptions
+    )
+  }
+}
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+extension Echo_EchoAsyncClientProtocol {
```

Comment:
I'm a bit hesitant to put the implementation of the simple-safe wrappers in the generated code for a couple of reasons:

1. It stops users from using the simple-safe wrappers who aren't using generated code (not using generated code is not all that rare)
2. Update are driven by the user regenerating their generated code

I think we add extensions to `GRPCClient` and call that in the generated code.

Reply:
Hm, OK. How would you like these spelled?

Reply:
I have spelled these `perform<type>Call` and made use of them in codgen in the update to this PR.

Reply:
Also... I added the extensions to `GRPCClient` but I also see we have a bunch of extensions on `GRPCChannel`. Do we want these wrappers to be on the channel API too?

Reply:
Apologies -- I had a comment written but didn't publish it:

---

Good question; we're not returning `Async{...}Call`s any more so `makeAsync{...}Call` certainly isn't right (and in some cases we'd be overloading on the return type which is a very bad idea).

Nothing jumps out at me. A couple of ideas though:
- `invoke{Unary....}RPC(...)`
- `invoke{Unary,...}Call(...)`
- `invoke{Unary....}(...)`

The could instead be called from wrapper struct with "friendlier" API. This is nice in that it separates the safe and unsafe calls from 

```swift
extension GRPCChannel {
  // Names are placeholders...
  public var asyncWrappers: GRPCAsyncWrappers { 
    return GRPCAsyncWrappers(self)
  }
}

public struct GRPCAsyncWrappers {
  public func unary<Request, Response>(...) async throws -> Response { ... }
  public func clientStreaming<Request, Response>(...) async throws -> Response { ... }
  // ... etc
}
```

Reply:
> Also... I added the extensions to `GRPCClient` but I also see we have a bunch of extensions on `GRPCChannel`. Do we want these wrappers to be on the channel API too?

Good spot; we probably don't actually need or want the extensions on `GRPCChannel`.

Reply:
> I have spelled these `perform<type>Call` and made use of them in codgen in the update to this PR.

I don't love it (or any of the other names I proposed) but it's fine for now.